### PR TITLE
feat(VDataTable) add dblclick:row event for data table

### LINKED ISSUE #10309: [Feature Request] v-data-table - add row double clicked event
### Problem to solve
Right now `v-data-table` has ```click:row``` event but it does not have ```dblclick:row``` event.

### Proposed solution
Add event which is triggered when we double click the row in `v-data-table`.

### PR DESCRIPTION
**Pull request by Leespiker**
## Description

add dblclick:row event for data table

## Motivation and Context

resolves #10309 

## How Has This Been Tested?

unit & visually

## Markup:

<details>

```vue
<template>
  <v-container>
    <v-data-table
      class="elevation-1"
      :headers="headers"
      :items="desserts"
      item-key="name"
      @dblclick:row="onRowDblClick"
    />

    <v-menu
      v-model="showMenu"
      :position-x="menuX"
      :position-y="menuY"
      absolute
      offset-y
    >
      <v-list>
        <v-list-item
          v-for="item in items"
          :key="item.title"
          @click=""
        >
          <v-list-item-title>{{ item.title }}</v-list-item-title>
        </v-list-item>
      </v-list>
    </v-menu>
  </v-container>
</template>

<script>
  export default {
    data: () => ({
      desserts: [
        {
          name: 'Frozen Yogurt',
          calories: 159,
          fat: 6.0,
          carbs: 24,
          protein: 4.0,
          iron: '1%',
        },
        {
          name: 'Ice cream sandwich',
          calories: 237,
          fat: 9.0,
          carbs: 37,
          protein: 4.3,
          iron: '1%',
        },
        {
          name: 'Eclair',
          calories: 262,
          fat: 16.0,
          carbs: 23,
          protein: 6.0,
          iron: '7%',
        },
        {
          name: 'Cupcake',
          calories: 305,
          fat: 3.7,
          carbs: 67,
          protein: 4.3,
          iron: '8%',
        },
        {
          name: 'Gingerbread',
          calories: 356,
          fat: 16.0,
          carbs: 49,
          protein: 3.9,
          iron: '16%',
        },
        {
          name: 'Jelly bean',
          calories: 375,
          fat: 0.0,
          carbs: 94,
          protein: 0.0,
          iron: '0%',
        },
        {
          name: 'Lollipop',
          calories: 392,
          fat: 0.2,
          carbs: 98,
          protein: 0,
          iron: '2%',
        },
        {
          name: 'Honeycomb',
          calories: 408,
          fat: 3.2,
          carbs: 87,
          protein: 6.5,
          iron: '45%',
        },
        {
          name: 'Donut',
          calories: 452,
          fat: 25.0,
          carbs: 51,
          protein: 4.9,
          iron: '22%',
        },
        {
          name: 'KitKat',
          calories: 518,
          fat: 26.0,
          carbs: 65,
          protein: 7,
          iron: '6%',
        },
      ],
      headers: [
        {
          align: 'start',
          sortable: false,
          text: 'Dessert (100g serving)',
          value: 'name',
        },
        {
          text: 'Calories',
          value: 'calories'
        },
        {
          text: 'Fat (g)',
          value: 'fat'
        },
        {
          text: 'Carbs (g)',
          value: 'carbs'
        },
        {
          text: 'Protein (g)',
          value: 'protein'
        },
        {
          text: 'Iron (%)',
          value: 'iron'
        },
      ],
      items: [],
      menuX: 0,
      menuY: 0,
      showMenu: false,
    }),
    methods: {
      onRowDblClick (event, data) {
        event.preventDefault()
        this.items = Object.entries(data).map(([key, value]) => {
          return {
            title: `${key}: ${typeof value === 'function' ? value.toString() : JSON.stringify(value)}`
          }
        }).sort((item1, item2) => item1.title.localeCompare(item2.title))
        this.showMenu = false
        this.menuX = event.clientX
        this.menuY = event.clientY
        this.$nextTick(() => {
          this.showMenu = true
        })
      }
    },
  }
</script>
```
</details>

## Types of changes

- [X] New feature (non-breaking change which adds functionality)

## Checklist:

- [X] The PR title is no longer than 64 characters.
- [X] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [X] My code follows the code style of this project.
- [X] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VDataTable/__tests__/VDataTable.spec.ts [hunk 1]
```diff
@@ -348,6 +348,24 @@ describe('VDataTable.ts', () => {
     expect(fn).toHaveBeenCalled()
   })
 
+  it('should emit event when double-clicking on internally created row', async () => {
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VDataTable/__tests__/VDataTable.spec.ts:351]: that's the same test as for `click:row` and (incoming) `contextmenu:row`, we could make it dry by iterating over `['click', 'dblclick', 'contextmenu']` array

Comment (Leespiker) [reply packages/vuetify/src/components/VDataTable/__tests__/VDataTable.spec.ts:351]: Good suggestion. I was thinking that too.

Comment (Leespiker) [reply packages/vuetify/src/components/VDataTable/__tests__/VDataTable.spec.ts:351]: I'll do that with whichever PR goes in last.

Comment (KaelWD) [reply packages/vuetify/src/components/VDataTable/__tests__/VDataTable.spec.ts:351]: https://jestjs.io/docs/en/api#describeeachtablename-fn-timeout

File: .gitignore [hunk 2]
```diff
@@ -6,6 +6,7 @@ node_modules/
 .vscode/
 .idea/
 .vs/
+*.iml
```
Comment (jacekkarczmarczyk) [inline .gitignore:9]: it doesn't look like it belongs to this PR

**Comment by CobyYates** [comment]
Is there a timeline for this feature being available?

**Comment by jacekkarczmarczyk** [comment]
it's already available in 2.3 beta

**Comment by CobyYates** [comment]
Just updated to beta 2.3 and the @dblclick:row="editItem" works with opening a dialog box, however the data doesn't pass to the editItem method like the @click:row="editItem" does. The first object in the payload of @click:row is the item that was selected and the first object for the @dblclick is the MouseEvent. Is there a workaround for passing the item? Also there a better place to post this question for feedback?

**Comment by diavrank** [comment]
Hi, I recommend to document this prop in https://vuetifyjs.com/en/components/data-tables/#api , since I think it has been requested by a lot of programmers.

**Comment by jacekkarczmarczyk** [comment]
It's not a prop, it's an event and it's mentioned in API
image

**Pull request merged**